### PR TITLE
Fix: in case of oauth token revoked, stop the connector instead of terminating the workflow

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

This previous [PR](https://github.com/dust-tt/dust/pull/6942) terminated a child workflow if the oauth token was revoked instead of cancelling it to prevent false positive in monitoring.

However, if a child workflow is `terminated`, the parent workflow is considered `failed` anyways so it didn't work as expected.

This new proposal uses `stop()` on the `ConnectorManager` to (hopefully) stop it properly.

Note that we could also call `terminateAllWorkflowsForConnectorId()` but it seems more appropriate to leverage the dedicated method. For example, the confluence one `stop()` method retrieve the parent workflow and terminate that one in particular.

## Risk

None

## Deploy Plan

Deploy `connectors`

### CODE REVIEW

**Comment by spolu** [comment]
Stopping will also stop the parent workflow. What's the difference?

**Review by spolu — APPROVED** [review]
So we went from cancel to terminate to stopping the connector which will have a custom implementation.

My only worry is if stop on the ConnectorManager removes the errorType for some reason in which case we end up with a stopped connector without errorType which is undesirable.

This would be caught by the check we added this week but would take a week to do so.

All that being said, looking at various stop implementations, I presume this is OK

File: connectors/src/lib/temporal_monitoring.ts [hunk 1]
```diff
@@ -121,11 +123,21 @@ export class ActivityInboundLogInterceptor
...
+          // In case of an invalid token, stop all workflows for the connector.
           this.logger.info(
-            `Terminating workflow ${workflowId} because of expired token.`
+            `Stopping connector manager because of expired token.`
           );
-          await terminateWorkflow(workflowId, "oauth_token_revoked");
+
+          const connector = await ConnectorResource.fetchById(connectorId);
+
+          if (connector) {
```
[the remark this answers is not in the document]
Comment (Fraggle) [reply connectors/src/lib/temporal_monitoring.ts:133]: Agree.

**Comment by Fraggle** [comment]
> Stopping will also stop the parent workflow. What's the difference?
> So we went from cancel to terminate to stopping the connector which will have a custom implementation.

That's the thing. Each connector might have a slightly different implementation.
When exploring the "Terminating all workflows" option, I wondered if order would matter and trigger the alert in certains cases. `stop()` is the idiomatic way to express the intent which then rely on a properly implemented connector.

> My only worry is if stop on the ConnectorManager removes the errorType for some reason in which case we end up with a stopped connector without errorType which is undesirable.

You are talking about the "syncFailed" errorType ? I'll take a look as well.

**Pull request merged**